augur/filter.py

```diff
@@ -430,3 +446,72 @@ def numeric_date(date):
         return float(date)
     except ValueError:
         return treetime.utils.numeric_date(datetime.date(*map(int, date.split("-", 2))))
+
+
+class TooManyGroupsError(ValueError):
+    ...
+
+
+def _calculate_total_sequences(
+        hypothetical_spg: int, sequence_lengths: Collection[int],
+) -> int:
+    # calculate how many sequences we'd keep given a hypothetical spg.
+    return sum(
+        min(hypothetical_spg, sequence_length)
+        for sequence_length in sequence_lengths
+    )
+
+
+def _calculate_spg(
+        target_max_value: int,
+        sequence_lengths: Collection[int],
+):
```

Comment:
Thank you for adding type hinting to these new functions! Would it make sense to include a type hint of `int` for the return value of this function, too?

Reply:
Will do.